### ISSUE TITLE
Possible prototype pollution with lodash.set

### ISSUE THREAD

**Issue by mausworks** [auth0/express-jwt#271]
### Description

As per the [vulnerability details on Snyk](https://security.snyk.io/vuln/SNYK-JS-LODASHSET-1320032), `lodash.set` which is used by this library allows for prototype pollution. This is not a major issue as this configuration is not controlled by the user, but by the configuring of the middleware—hence why I report this here, and not via the vulnerability disclosure program.

We believe that the best remediation of this is to stop using the `lodash.set` package, and instead use the [mainline `lodash` package](https://www.npmjs.com/package/lodash) as [it has been remediated there in v > 4.17.17](https://security.snyk.io/vuln/SNYK-JS-LODASH-608086), and because the `lodash.set` package has not been published for five years on NPM.

You can require `set` explicitly with `const set = require('lodash/set');`.

### Reproduction

Configure the middleware with:

```
jwt({
  secret: 'shhhhhhared-secret',
  audience: 'http://myapi/protected',
  issuer: 'http://issuer',
  algorithms: ['HS256'],
  userProperty: '__proto__[test]'
})
```

... and you will pollute the prototype of the request object.

### Environment

- Version: Any version of express-jwt that uses lodash.set
- Other involved packages: `lodash.set`

**Comment by sirgoofiguskid** [comment]
Should probably use lodash 4.17.21, as earlier packages have a variety of other issues: https://snyk.io/vuln/npm%3Alodash

**Comment by ryanpcmcquen** [comment]
PR:
https://github.com/auth0/express-jwt/pull/272

**Comment by bjornstar** [comment]
This looks to be fixed in release v6.1.1
